### ISSUE TITLE
Fix Request: Kotlin AvoidRecreatingDateTimeFormatter false positive

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#669]
The rule doesn't recognize `object` as being a singleton and `val` for a constant.
```
object DateUtil {
    private const val ISO8601_PATTERN = "yyyyMMdd'T'HHmmss'Z'"
    private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(ISO8601_PATTERN) // <-- false positive

    fun formatISO8601(dateTime: ZonedDateTime): String =
        dateTime.withZoneSameInstant(ZoneOffset.UTC)
            .format(DATE_TIME_FORMATTER)
}
```

**Comment by stokpop** [comment]
Also for these contructs, format can also be on String (and other types?):

```kotlin
class Foo() {
    fun originalDate(rule: Rule): String? =
        rule.date?.format(DateTimeFormatter.ISO_DATE) // <-- false positive
}

class Rule(val name: String, val regex: String, val date: LocalDate?) 
```
